homeassistant/components/blink/config_flow.py

```diff
@@ -0,0 +1,119 @@
+    CONF_PIN,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+
+from .const import DEFAULT_OFFSET, DEFAULT_SCAN_INTERVAL, DEVICE_ID, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(hass: core.HomeAssistant, blink, data):
+    """Validate the user input allows us to connect."""
+    response = await hass.async_add_executor_job(blink.get_auth_token)
+    if not response:
+        raise InvalidAuth
+    if blink.key_required:
+        raise Require2FA
+
+    return blink.login_response
+
+
+class BlinkConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Blink config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize the blink flow."""
+        self.blink = None
+        self.data = {
+            CONF_USERNAME: "",
+            CONF_PASSWORD: "",
+            CONF_SCAN_INTERVAL: DEFAULT_SCAN_INTERVAL,
+            "login_response": None,
+        }
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        errors = {}
+        if user_input is not None:
+            self.data[CONF_USERNAME] = user_input["username"]
+            self.data[CONF_PASSWORD] = user_input["password"]
+
+            await self.async_set_unique_id(self.data[CONF_USERNAME])
+
+            if CONF_SCAN_INTERVAL in user_input:
+                self.data[CONF_SCAN_INTERVAL] = user_input["scan_interval"]
+
+            self.blink = Blink(
+                username=self.data[CONF_USERNAME],
+                password=self.data[CONF_PASSWORD],
+                motion_interval=DEFAULT_OFFSET,
+                legacy_subdomain=False,
+                no_prompt=True,
+                device_id=DEVICE_ID,
+            )
+
+            try:
+                response = await validate_input(self.hass, self.blink, self.data)
+                self.data["login_response"] = response
+                return self.async_create_entry(title=DOMAIN, data=self.data,)
+            except Require2FA:
+                return await self.async_step_2fa()
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+
+        data_schema = {
+            vol.Required("username"): str,
+            vol.Required("password"): str,
+        }
+
+        if self.show_advanced_options:
+            data_schema[
+                vol.Required("scan_interval", default=DEFAULT_SCAN_INTERVAL)
+            ] = int
+
+        return self.async_show_form(
+            step_id="user", data_schema=vol.Schema(data_schema), errors=errors,
+        )
+
+    async def async_step_2fa(self, user_input=None):
+        """Handle 2FA step."""
+        if user_input:
+            pin = user_input[CONF_PIN]
+            if not pin or pin.lower() == "none":
```

Comment:
When will the pin be `"None"`?

Reply:
Oh. So that's the official instructions from Blink?

Reply:
If the library accepts `None`, why can't we make the pin optional in the form?

Reply:
Why do we need to do an extra check? `None` is accepted by the library, right? So just pass the pin forward to the library.

Reply:
If we want to guard against empty string we can add a `vol.Length` check to the form data schema.
```py
data_schema = vol.Schema({vol.Required("pin"): vol.All(str, vol.Length(min=1))})
```

Reply:
See my suggested changes below. I think that should work. Please test it.